### PR TITLE
Feature/sm 34 user models

### PR DESCRIPTION
**Pull request by Sashimi4**

### CODE REVIEW

File: backend/src/main/java/com/social/dailylink/controllers/AuthController.java [hunk 1]
```diff
@@ -93,26 +94,28 @@ public ResponseEntity<?> registerUser(@Valid @RequestBody SignupRequest signUpRe
     Set<String> strRoles = signUpRequest.getRole();
     Set<Role> roles = new HashSet<>();
 
+    final String errorRoleString = "Error: Role is not found.";
```
Comment (davidemarcoli) [inline backend/src/main/java/com/social/dailylink/controllers/AuthController.java]: Constants should be all Uppercase

**Review by davidemarcoli — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/social/dailylink/controllers/AuthController.java (hunk 1 above)
Comment (Sashimi4) [reply backend/src/main/java/com/social/dailylink/controllers/AuthController.java]: Acknowledged

File: backend/src/main/java/com/social/dailylink/controllers/AuthController.java [hunk 2]
```diff
@@ -94,28 +95,26 @@ public ResponseEntity<?> registerUser(@Valid @RequestBody SignupRequest signUpRe
     Set<String> strRoles = signUpRequest.getRole();
     Set<Role> roles = new HashSet<>();
 
-    final String errorRoleString = "Error: Role is not found.";
-
     if (strRoles == null) {
       Role userRole = roleRepository.findByName(ERole.ROLE_USER)
-          .orElseThrow(() -> new RuntimeException(errorRoleString));
+          .orElseThrow(() -> new RuntimeException(Global.GLOBAL_STRINGS.getERROR_ROLE_NOT_FOUND()));
```
Comment (davidemarcoli) [inline backend/src/main/java/com/social/dailylink/controllers/AuthController.java]: Strange getters! Are they needed? Can't we use static variables if they are final?

File: backend/src/main/java/com/social/dailylink/global/GlobalStrings.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    public static GlobalStrings getInstance() {
+        if(INSTANCE == null) {
+            INSTANCE = new GlobalStrings();
+        }
+        return INSTANCE;
+    }
+
+    public String getSchemaName() {
```
Comment (davidemarcoli) [inline backend/src/main/java/com/social/dailylink/global/GlobalStrings.java]: Are these needed?

**Review by davidemarcoli — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/social/dailylink/controllers/AuthController.java (hunk 2 above)
Comment (lazarpe) [reply backend/src/main/java/com/social/dailylink/controllers/AuthController.java]: +1

File: dailylink.iml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  <component name="NewModuleRootManager" LANGUAGE_LEVEL="JDK_1_8">
+    <output url="file://$MODULE_DIR$/target/classes" />
+    <output-test url="file://$MODULE_DIR$/target/test-classes" />
+    <content url="file://$MODULE_DIR$">
+      <excludeFolder url="file://$MODULE_DIR$/target" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (lazarpe) [inline dailylink.iml:12]: Please add a newline at end of .iml files

File: frontend/frontend.iml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  <component name="NewModuleRootManager" LANGUAGE_LEVEL="JDK_1_8">
+    <output url="file://$MODULE_DIR$/target/classes" />
+    <output-test url="file://$MODULE_DIR$/target/test-classes" />
+    <content url="file://$MODULE_DIR$">
+      <excludeFolder url="file://$MODULE_DIR$/target" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (lazarpe) [inline frontend/frontend.iml]: Same here. Please add a newline at the end.

**Review by lazarpe — DISMISSED** [review]

File: backend/src/main/java/com/social/dailylink/controllers/AuthController.java [hunk 6]
```diff
@@ -48,6 +49,7 @@ public class AuthController {
   @PostMapping("/signin")
   public ResponseEntity<?> authenticateUser(@Valid @RequestBody LoginRequest loginRequest) {
 
+    //Must be replaced with Logger
```
Comment (davidemarcoli) [inline backend/src/main/java/com/social/dailylink/controllers/AuthController.java]: Add TODO Comments, not plain comments. They aren't highlighted when looking at a project in an IDE

File: dailylink.iml (hunk 4 above)
Comment (davidemarcoli) [reply dailylink.iml:12]: Could be solved with a formatter

**Review by davidemarcoli — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/social/dailylink/global/GlobalStrings.java (hunk 3 above)
Comment (Sashimi4) [reply backend/src/main/java/com/social/dailylink/global/GlobalStrings.java]: No they are not.

File: dailylink.iml (hunk 4 above)
Comment (Sashimi4) [reply dailylink.iml:12]: I will ignore this for now.

File: frontend/frontend.iml (hunk 5 above)
Comment (Sashimi4) [reply frontend/frontend.iml]: Will add in future change.

**Review by davidemarcoli — APPROVED** [review]
Now it looks good to me!

**Review by davidemarcoli — APPROVED** [review]

**Pull request merged**